lib/vfs/src/mem_fs/filesystem.rs

```diff
@@ -605,8 +605,8 @@ impl FileSystemInner {
                     Some(Node::Directory(DirectoryNode { .. })) => {
                         Ok(InodeResolution::Found(inode_of_parent))
                     }
-                    Some(Node::ArcDirectory(ArcDirectoryNode { .. })) => {
-                        Ok(InodeResolution::Found(inode_of_parent))
+                    Some(Node::ArcDirectory(ArcDirectoryNode { fs, path, .. })) => {
+                        Ok(InodeResolution::Redirect(fs.clone(), path.clone()))
```

Comment:
Nice find. 

Let's either add a small unit test for this here, or create a follow up issue to add one.

Reply:
There isn't a single test with `ArcDirectory`. Not sure when this has been introduced, but it is test-less

Reply:
Added a mention of this is ticket https://github.com/wasmerio/wasmer/issues/3678 similar this.

Reply:
I'll add an explicit issue, because otherwise we will forget.